contractsSol5/Dao/KyberFeeHandler.sol

```diff
@@ -196,21 +202,29 @@ contract KyberFeeHandler is IKyberFeeHandler, Utils4, BurnConfigPermission {
             return true;
         }
 
+        WeiData memory weiData;
+        uint epoch;
         // Decoding BRR data
-        (uint rewardWei, uint rebateWei, uint epoch) = getRRWeiValues(feeBRR);
+        (weiData.reward, weiData.rebate, epoch) = getRRWeiValues(feeBRR);
 
+        uint totalRebateBps;
         for (uint i = 0; i < rebateWallets.length; i++) {
             // Internal accounting for rebates per reserve wallet (rebatePerWallet)
-            rebatePerWallet[rebateWallets[i]] += rebateWei * rebateBpsPerWallet[i] / BPS;
+            require(rebateWallets[i] != address(0), "rebate wallet 0");
+            rebatePerWallet[rebateWallets[i]] += weiData.rebate * rebateBpsPerWallet[i] / BPS;
+            // a few wei could be left out due to rounding down. so count only paid wei
+            weiData.totalRebatePaid += weiData.rebate * rebateBpsPerWallet[i] / BPS;
+            totalRebateBps += rebateBpsPerWallet[i];
```

Comment:
Assume we trust the data passing from network here? i.e `rebateBpsPerWallet[i] <= 100%`